src/copy.rs

```diff
@@ -0,0 +1,377 @@
+impl<'a, A: Array + Copy> IntoIterator for &'a ArrayVecCopy<A> {
+    type Item = &'a A::Item;
+    type IntoIter = slice::Iter<'a, A::Item>;
+    fn into_iter(self) -> Self::IntoIter { self.inner.iter() }
+}
+
+/// Iterate the `ArrayVecCopy` with mutable references to each element.
+impl<'a, A: Array + Copy> IntoIterator for &'a mut ArrayVecCopy<A> {
+    type Item = &'a mut A::Item;
+    type IntoIter = slice::IterMut<'a, A::Item>;
+    fn into_iter(self) -> Self::IntoIter { self.inner.iter_mut() }
+}
+
+/// Iterate the `ArrayVecCopy` with each element by value.
+///
+/// The vector is consumed by this operation.
+impl<A: Array + Copy> IntoIterator for ArrayVecCopy<A> {
+    type Item = A::Item;
+    type IntoIter = IntoIter<A>;
+    fn into_iter(self) -> IntoIter<A> {
+        IntoIter { index: Index::from(0), v: self }
+    }
+}
+
+
+/// By-value iterator for `ArrayVecCopy`.
+pub struct IntoIter<A: Array + Copy> {
+    index: A::Index,
+    v: ArrayVecCopy<A>,
+}
+
+impl<A: Array + Copy> Iterator for IntoIter<A> {
+    type Item = A::Item;
+
+    #[inline]
+    fn next(&mut self) -> Option<A::Item> {
+        let index = self.index.to_usize();
+        if index == self.v.len() {
+            None
+        } else {
+            unsafe {
+                self.index = Index::from(index + 1);
+                Some(ptr::read(self.v.get_unchecked_mut(index)))
+            }
+        }
+    }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = self.v.len() - self.index.to_usize();
+        (len, Some(len))
+    }
+}
+
+impl<A: Array + Copy> DoubleEndedIterator for IntoIter<A> {
+    #[inline]
+    fn next_back(&mut self) -> Option<A::Item> {
+        if self.index.to_usize() == self.v.len() {
+            None
+        } else {
+            unsafe {
+                let new_len = self.v.len() - 1;
+                self.v.set_len(new_len);
+                Some(ptr::read(self.v.get_unchecked_mut(new_len)))
+            }
+        }
+    }
+}
+
+impl<A: Array + Copy> ExactSizeIterator for IntoIter<A> { }
+
+/// Extend the `ArrayVecCopy` with an iterator.
+/// 
+/// Does not extract more items than there is space for. No error
+/// occurs if there are more iterator elements.
+impl<A: Array + Copy> Extend<A::Item> for ArrayVecCopy<A> {
+    fn extend<T: IntoIterator<Item=A::Item>>(&mut self, iter: T) {
+        self.inner.extend(iter)
+    }
+}
+
+/// Create an `ArrayVecCopy` from an iterator.
+/// 
+/// Does not extract more items than there is space for. No error
+/// occurs if there are more iterator elements.
+impl<A: Array + Copy> iter::FromIterator<A::Item> for ArrayVecCopy<A> {
+    fn from_iter<T: IntoIterator<Item=A::Item>>(iter: T) -> Self {
+        ArrayVecCopy { inner: RawArrayVec::from_iter(iter) }
+    }
+}
+
+impl<A: Array + Copy> Clone for ArrayVecCopy<A>
+    where A::Item: Clone
+{
+    #[inline]
+    fn clone(&self) -> Self {
+        ArrayVecCopy { inner: self.inner.clone() }
```

Comment:
Yea, but this is already the case for any new `ArrayVec`, all the elements are uninitialized there.